### PR TITLE
feat(cmf): add react-cmf-webpack-plugin

### PR DESCRIPTION
**Pull request by frassinier**
**What is the problem this PR is trying to solve?**
We have to run a script each times we want to merge several settings

**What is the chosen solution to this problem?**
New Webpack plugin to do it automatically

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

**How to test**

In order to test this plugin, you have to:

1. `cd <TALEND_UI_ROOT_PATH>`
1. `git fetch && git checkout frassinier/feat/cmf-webpack-plugin`
1. `cd packages/cmf-webpack-plugin && yarn && yarn link`
1. `cd <YOUR_WEBAPP_ROOT_PATH> && yarn link @talend/react-cmf-webpack-plugin`
1. Edit your webpack config according to [this file](https://github.com/Talend/ui/blob/frassinier/feat/cmf-webpack-plugin/packages/cmf-webpack-plugin/README.md)
1. Be sure to have a _cmf.json_ file
1. `yarn start`

### CODE REVIEW

File: packages/cmf-webpack-plugin/README.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+		],
+		"destination": "dist/settings.json"
+	}
+}
+```
+
+Edit your webpack.config.js file:
+
+```javascript
+const ReactCMFWebpackPlugin = require('react-cmf-webpack-plugin');
```
Comment (jmfrancois) [inline packages/cmf-webpack-plugin/README.md]: @talend/react-cmf-webpack-plugin

File: packages/cmf-webpack-plugin/README.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+[![NPM][npm-icon] ][npm-url]
+
+[![Travis CI][travis-ci-image] ][travis-ci-url]
+[![Quality][quality-badge] ][quality-url]
+[![dependencies][dependencies-image] ][dependencies-url]
+[![devdependencies][devdependencies-image] ][devdependencies-url]
+[![Codacy Badge](https://api.codacy.com/project/badge/Grade/f1e353b0f69c4cf99a4cb3f68b70ea7d)](https://www.codacy.com/app/Talend/ui/packages/cmf-cqrs)
+
+[npm-icon]: https://nodei.co/npm/react-cmf-webpack-plugin.png?downloads=true
```
Comment (jmfrancois) [inline packages/cmf-webpack-plugin/README.md]: need @talend

File: packages/cmf-webpack-plugin/README.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+| --- | --- | --- | --- |
+| `dev` | Boolean | false | `devSource` entry will be used instead of `sources` one in _cmf.json_ file |
+| `quiet` | Boolean | false | No output at all |
+| `recursive` | Boolean | false | Recursive search for JSON files |
+| `watch` | Boolean | false | Watch settings in dev mode |
+
+Here's an example webpack config illustrating how to use these options:
+
+```javascript
+const ReactCMFWebpackPlugin = require('react-cmf-webpack-plugin');
```
Comment (jmfrancois) [inline packages/cmf-webpack-plugin/README.md]: need @talend

File: packages/cmf-webpack-plugin/package.json [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    "framework"
+  ],
+  "author": "Talend Frontend <frontend@talend.com>",
+  "license": "Apache-2.0",
+  "bugs": {
+    "url": "https://github.com/Talend/ui/issues"
+  },
+  "homepage": "https://github.com/Talend/ui/blob/master/packages/cmf/README.md",
+  "dependencies": {
+    "@talend/react-cmf": "file:../cmf",
```
Comment (jsomsanith) [inline packages/cmf-webpack-plugin/package.json]: Please replace by version. It is part of yarn workspace, so the links are done automatically

File: packages/cmf-webpack-plugin/src/index.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+	atomic: true,
+};
+
+/**
+ * React CMF Webpack Plugin
+ * @param options Plugin options
+ * @constructor
+ */
+function ReactCMFWebpackPlugin(options = {}) {
+	this._canRun = true;
```
Comment (jsomsanith) [inline packages/cmf-webpack-plugin/src/index.js]: Why do you have `_` on all variables ? This makes linter goes wild

**Pull request merged**